src/main/java/classify/commands/DeleteCommands.java

```diff
@@ -114,6 +114,21 @@ public static void undoDelete(ArrayList<Student> masterStudentList, ArrayList<St
             return;
         }
         Student student = recentlyDeletedList.get(recentlyDeletedList.size() - 1);
+
```

Comment:
Same checks that have been implemented in restore, in the case that save file is corrupted/edited. LGTM!